### DISCUSSION TITLE
More builtins for Hilbish

### DISCUSSION THREAD

**Discussion by sammy-ette** [sammy-ette/Hilbish#44] [Ideas]
Hilbish is steadily coming close to a stable point where I'm not even working on it consistently anymore. So, to alleviate that I'm asking for any ideas on extra command built ins or global functions and things like that to Hilbish

or, what convenient functions you'd like, for example would you like `string.split()` to more easily split up strings?
*recently, the very convenient appendPath function was added :+1:*

**Comment by tembokk** [comment]
I think something like neovim's `vim.inspect` will be very useful

**Reply by sammy-ette** [reply]
i just included inspect.lua as a submodule, done

**Comment by nerd190** [comment]
zsh has some loadable functions for handling navigation,  the main one is `cdr()` (cd recent), its saves a list of your visited directories (default is 10) and gives a simple syntax (`cdr 2`) to navigate them, along with some configuration, you can also integrate it with zsh's `autocd` (bash has this too), `autopushd` (bash hasnt got this!) and more.

zsh also has `zmv`/`zcp`/`zln` (zsh move/copy/link) which accepts zsh-style globbing i.e
```sh
zmv '(**/)(*).txt '$1$2.md'
# Rename "foo.txt" to "foo.md" recursively.
zmv '([a-z])(*).txt' '${(C)1}$2.txt' 
# Renames "algernon.txt" to "Algernon.txt", "boris.txt" to "Boris.txt" etc.
```
As Lua has great regex support, something similar to this would be great!

**Reply by sammy-ette** [reply]
awesome! what i meant for builtins here was more or less lua api additions, but commands are good as well.
`cdr` seems to be pretty easy to make ~~but i dont know about `zmv` and the others. what would they be named in hilbish? should they even be separate commands?~~

also be sure when replying to this, dont write a separate comment to the discussion but reply to this comment u already posted

**Reply by sammy-ette** [reply]
added cdr
